stl/src/mutex.cpp

```diff
@@ -147,8 +147,9 @@ static _Thrd_result mtx_do_lock(_Mtx_t mtx, const _timespec64* target) noexcept
 }
 
 _CRTIMP2_PURE _Thrd_result __cdecl _Mtx_unlock(_Mtx_t mtx) noexcept { // unlock mutex
+    _THREAD_ASSERT(1 <= mtx->_Count, "unlock of unowned mutex");

```

Comment:
I'd prefer having `mtx->_Count > 0` here and in another debug check in `_Mtx_destroy_in_situ`.

`mtx->_Count == 0` is also a straightforward form, but `<` also checks the invariant of non-negative count for free.